File: common/src/main/java/com.codeabovelab.tpc/text/TextualImpl.java
package com.codeabovelab.tpc.text;

/**
 */
public final class TextualImpl implements Textual {
    private final String id;
    private final TextImpl text;

    public TextualImpl(String id, String text) {
        this.id = id;
        this.text = new TextImpl(this.id, text);
    }

    @Override
    public String getId() {
        return id;
    }

    @Override
    public void read(TextConsumer consumer) {
        consumer.consume(text);
    }
}

File: common/src/main/java/com.codeabovelab.tpc/text/TextualImpl.kt
package com.codeabovelab.tpc.text

/**
 * Simple implementation of textual.
 */
class TextualImpl(
        override val id: String,
        override val data: String,
        override val parent: Textual?
) : Textual, Text {

    override val length: Int get() = data.length

    override fun read(consumer: TextConsumer) {
        consumer(this, this)
    }
}
